### PR TITLE
Report available services in the user profile action

### PR DESCRIPTION
**Pull request by aub**
The user profile JSON response should include information about which services are available to the user so that the client-side code can block them from accessing services that are unavailable. This fills in that block with the array of available services. I've added new methods to the User class for `can_access_*` for the three services in question: user_profile, mhv, and evss. Then, the user serializer uses those three methods to inform the list of available services. The user profile service is active for all LOA levels, so I added query methods for those as well, but that feels a bit janky. If anybody has suggestions on a better way to do this, fire away.

### CODE REVIEW

**Comment by robbiethegeek** [comment]
Looks good, @omgitsbillryan please take a look /cc @crwallace

**Comment by aub** [comment]
Except for my boneheaded spec failures.

**Comment by saneshark** [comment]
@aub any reason why we prefer this vs cancancan or pundit for authorization? Both are easy to customize and provide a number of helper and convenient methods. They may be somewhat heavy-handed for this purpose though.

It was discussed in the past that we may eventually have OAuth 2.0 scopes, which I think will work seamlessly with pundit or cancancan if I recall. That being said, it's probably something we can revisit later.

**Comment by aub** [comment]
@saneshark It's that cancancan was deemed to be too heavyweight for what we needed here.

File: app/serializers/user_serializer.rb [hunk 1]
```diff
@@ -31,7 +31,10 @@ def va_profile
   end
 
   def services
-    # TODO: build the available services based on available ID's
-    []
+    %w(facilities hca edu-benefits).tap do |service_list|
+      service_list += %w(rx messaging) if object.can_access_mhv?
```
Comment (omgitsbillryan) [inline app/serializers/user_serializer.rb]: Would be nice to have rx, messaging, disability-benefits, & user-profile as constants defined somewhere and then re-used in the rspec.  That way, rspecs wont need to be updated if the values change.

Comment (aub) [inline app/serializers/user_serializer.rb]: That makes sense. Anybody have thoughts on where those constants ought to be defined?

**Comment by aub** [comment]
@omgitsbillryan I've moved the service names out into constants. I looked at the initializer code you mentioned, but I don't think it makes sense to use an initializer in this case because it's neither initializing some service nor loading the data from configuration, so I created a class in /lib to hold the constants.

**Pull request merged**